python/llm/example/CPU/Applications/autogen/README.md

```diff
@@ -0,0 +1,236 @@
+## Running AutoGen Agent Chat with BigDL-LLM on Local Models
+In this example, we use BigDL adapted FastChat to run [AutoGen](https://microsoft.github.io/autogen/) agent chat with 
+local large language models.
+
+### 1. Prepare Environment
+```bash
+# create autogen running directory
+mkdir autogen
+cd autogen
+
+# create respective conda environment
+conda create -n autogen python=3.9
+conda activate autogen
+
+# install fastchat-adapted bigdl-llm
+pip install --pre --upgrade bigdl-llm[serving]==2.5.0b20240110
+
+# clone the BigDL in the autogen folder
+git clone https://github.com/intel-analytics/BigDL.git
+
+# install recommend transformers version
+pip install transformers==4.36.2
+
+# install necessary dependencies
+pip install chromadb==0.4.22
+```
+
+
+### 2. Setup FastChat and AutoGen Environment
+```bash
+# clone the FastChat in the autogen folder
+git clone https://github.com/lm-sys/FastChat.git FastChat # clone the FastChat
+cd FastChat
+pip3 install --upgrade pip  # enable PEP 660 support
+pip3 install -e ".[model_worker,webui]"
+
+# setup AutoGen environment
+pip install pyautogen==0.2.7
+```
+
+### 3. Build FastChat OpenAI-Compatible RESTful API
+Open 3 terminals
+
+**Terminal 1: Launch the controller**
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the cloned FastChat folder in autogen folder
+cd autogen/FastChat
+
+python -m fastchat.serve.controller
+```
+
+**Terminal 2: Launch the workers**
+
+```bash
+# activate conda environment
+conda activate autogen
+
+# go to the created autogen folder
+cd autogen
+
+# ensure the connection between controller and worker
+export no_proxy='localhost'
```

Comment:
no need to put this in the script. Add a note in README and tell user if they're using this under a proxy, they may need to set this.